nuget/dotnet/benchmarks/BenchmarkCbAdfLearn.cs

```diff
@@ -0,0 +1,43 @@
+﻿using VW;
+using BenchmarkDotNet.Attributes;
+
+[Config(typeof(VWBenchmarkConfig))]
+public class BenchmarkCbAdfLearn
+{
+    [Benchmark]
+    [ArgumentsSource(nameof(ArgsGenerator))]
+    public void Benchmark(Args args)
+    {
+        args.vw.Learn(args.examples);
```

Comment:
Does finish example need to be called? I am not sure how this is setup, so just wondering

Reply:
It's called in the native DLL wrapper here: https://github.com/byronxu99/vowpal_wabbit/blob/dev/DotNetBenchmarks/cs/vw.net.native/vw.net.workspace.cc#L289

which is in turn called by C# code here: https://github.com/byronxu99/vowpal_wabbit/blob/dev/DotNetBenchmarks/cs/vw.net/VowpalWabbit.cs#L630

Reply:
This means that the managed example object is technically invalidated after making this call. It would be quite easy for a user to think they can reuse the example object. Feels a bit scary

Reply:
Looks like this is the same as how the old .NET Framework bindings did it: https://github.com/byronxu99/vowpal_wabbit/blob/0e503b39662ca14b64e3c887cb3799b6f50cf91e/cs/cli/vowpalwabbit.cpp#L212

I do agree that it seems a bit scary.

Reply:
Yeah, this is an unfortunate legacy of the code as it was. What I would like to do going forward is to formalize the proposal for the c interface that you started @jackgerrits, and use a C#-ified layer as the new canonical interface of VW.Net.